### DISCUSSION TITLE
Adding extra EXIF data such as Camera Make, Model, Lens etc

### DISCUSSION THREAD

**Discussion by rollbahn** [RhetTbull/osxphotos#1471] [Q&A] [answered]
Hi,
I am trying to add extra exif data and thought the –exiftool-option might allow me to sneak it through...even though I knew it wouldn't, it was worth a try 😀. I have a bunch of scanned film images that I would like to be able to add the correct make, model and lens data in Photos as I currently do via a plugin in Lightroom. However I am trying to move away from LR so I would like to be able to do this via osxphotos rather than making sure to remember to do it directly in exiftool before importing into Photos. 

Here's an idea of what I would like to run - is there a way this can be done or is it not supported right now?

osxphotos push-exif --exiftool-merge-keywords --person-keyword --exiftool-merge-persons –exiftool-option '-Make=Ricoh' –exiftool-option '-Model=GR1v' –exiftool-option '-Lens=Ricoh GR 28mm f/2.8' --name 00655_27_newscan.jpg

Thanks for your help.
Jason

**Comment by RhetTbull** [comment, marked as answer]
>  rather than making sure to remember to do it directly in exiftool before importing into Photos.

Unfortunately that's the only way to do this currently. 

The `--exiftool-option` is only for passing flags like `-m` to exiftool.

I do have an issue to add a `--exiftool-template` option that would allow you to do what you're stating  (#243) but that's not implemented yet and I'm currently busy with some other things so can't say when it'll be done.

In the meantime you could use the `osxphotos query` command to get the path to the file and pipe it to exiftool to run your command.

For example, for the currently selected photo(s):

`osxphotos query --quiet --selected --print "{photo.path}" | xargs -I {} exiftool {}`

This simply runs exiftool on the selected photo(s). It does not do what `push-exif` does so you'd still have to do that separately.

For your example above to work on just the ` 00655_27_newscan.jpg` image:

`osxphotos query --quiet --print "{photo.path}" --name  00655_27_newscan.jpg | xargs -I {} exiftool {} -Make=Ricoh -Model=GR1v -Lens=Ricoh GR 28mm f/2.8`

Please note that I *do not* recommend this if you are running an iCloud library as these changes will not be synced to iCloud and could be overwritten by iCloud. If using iCloud the only safe option is to export the images, apply the metadata then re-import.

Also note that this does not change the metadata stored in the Photos database so it won't appear in the `Info` window for the photo. This is only useful if you want to preserve the data in the original image for future export and you aren't using iCloud.

**Reply by rollbahn** [reply]
Thanks very much for the detailed answer - it is much appreciated. My main issue with using Photos over LR is the ability to be in control of my own data and know that I control the originals unlike Photos. So if I want to get my images out for any reason and know that the originals will have all my info is great.

It looks like I will have to make sure to add in camera/lens data etc before importing into Photos and then the rest of the exif I know I can tweak in Photos.

Thanks again for your time.
